### PR TITLE
Applications: update inline docs for `only_available` in applications addon settings

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
Add some inline docs for `only_available` setting in application addon.

## Additional info
This PR was made based on some community feedback.

## Testing notes:
1. Everything should remain the same..
2. Tell me if the added inline docs make sense.

### CODE REVIEW

File: server_addon/applications/server/settings.py [hunk 1]
```diff
@@ -144,7 +144,17 @@ def validate_unique_name(cls, value):
 
 
 class ApplicationsSettings(BaseSettingsModel):
-    """Applications settings"""
+    """Applications settings
+
+    Define applications are available to your studio, locations of their
+    executables, and their additional environments.
```
Comment (BigRoy) [inline server_addon/applications/server/settings.py:150]:
```suggestion
    Define applications available to your studio, locations of their
    executables, and their additional environments.
```
---
OR alternatives:

```suggestion
    Define applications that are available to your studio, locations of their
    executables, and their additional environments.
```
OR
```suggestion
    Define applications which are available to your studio, locations of their
    executables, and their additional environments.
```

File: server_addon/applications/server/settings.py [hunk 2]
```diff
@@ -144,7 +144,17 @@ def validate_unique_name(cls, value):
...
+
+    AYON uses the following definitions to populate the applications list
+    in project anatomy > attributes.
+    So, You'd still need modify your project anatomy settings.
+
+    Use shortcut **a+a** to navigate to project anatomy.
```
Comment (BigRoy) [inline server_addon/applications/server/settings.py:156]: I feel like front-end shortcuts should not be in the docstrings of the codebase - it'll just be too hard to keep in sync.
```suggestion
```

File: server_addon/applications/server/settings.py [hunk 3]
```diff
@@ -144,7 +144,17 @@ def validate_unique_name(cls, value):
...
 class ApplicationsSettings(BaseSettingsModel):
-    """Applications settings"""
+    """Applications settings
+
+    Define applications are available to your studio, locations of their
+    executables, and their additional environments.
+
+    AYON uses the following definitions to populate the applications list
+    in project anatomy > attributes.
+    So, You'd still need modify your project anatomy settings.
```
Comment (BigRoy) [inline server_addon/applications/server/settings.py:154]: This comment is unclear to me. What are you trying to explain here?

File: server_addon/applications/server/settings.py [hunk 4]
```diff
@@ -210,7 +220,14 @@ class ApplicationsAddonSettings(BaseSettingsModel):
...
     )
     only_available: bool = SettingsField(
-        True, title="Show only available applications")
+        True,
+        title="Show only available applications",
+        description="Enable to show available applications in AYON Launcher"
+                    " i.e. Show them if they are installed on the user"
+                    " machine. Note: AYON Launcher shows only the selected"
+                    " applications in project anatomy and this is another"
+                    " filter to ignore uninstalled apps."
```
Comment (BigRoy) [inline server_addon/applications/server/settings.py:229]: I feel like the _NOTE_ isn't that helpful. Maybe you like some parts of this re-write?
```suggestion
        description="Enable to show only available applications in AYON Launcher"
                    " for which the executable paths are found on the running machine."
                    " This applies as an additional filter to the applications defined in a "
                    " project's anatomy settings to ignore unavailable applications."
```

File: server_addon/applications/server/settings.py (hunk 1 above)
Comment (BigRoy) [reply server_addon/applications/server/settings.py:150]: Maybe also `additional environments` is better written as `additional environment variables` for full clarity?

File: server_addon/applications/server/settings.py (hunk 3 above)
Comment (BigRoy) [reply server_addon/applications/server/settings.py:154]: Ah, that makes sense. I have a hard time describing that without a perspective of what the anatomy settings are. 

```suggestion
    AYON uses the Applications Settings definition to build the
    list of all available applications in the studio settings list. 
    Those then define the application variants that can be 
    enabled per project in the project anatomy settings.
```

Not sure if that's any better. I feel like there must be a better way to describe this with not too many words, but I can't think of any.

Comment (MustafaJafar) [reply server_addon/applications/server/settings.py:154]: what about explaining the end result ?
```suggestion
Hint: 
    AYON Launcher will show the following applications list filtered by your project anatomy settings.
```

Comment (BigRoy) [reply server_addon/applications/server/settings.py:154]: I personally like that wording less - but maybe someone else can chime in.

File: server_addon/applications/server/settings.py [hunk 5]
```diff
@@ -212,7 +212,13 @@ class ApplicationsAddonSettings(BaseSettingsModel):
         scope=["studio"]
     )
     only_available: bool = SettingsField(
-        True, title="Show only available applications")
+        True,
+        title="Show only available applications",
+        description="Enable to show only available applications in AYON Launcher"
```
Comment (iLLiCiTiT) [inline server_addon/applications/server/settings.py:217]: I would not mention AYON launcher as webactions are comming soon...
```suggestion
        description="Enable to show only available applications"
```

File: server_addon/applications/server/settings.py (hunk 3 above)
Comment (MustafaJafar) [reply server_addon/applications/server/settings.py]: tagging @dee-ynput for information.

File: server_addon/applications/server/settings.py [hunk 6]
```diff
@@ -212,7 +212,13 @@ class ApplicationsAddonSettings(BaseSettingsModel):
...
+                    " for which the executable paths are found on the running machine."
+                    " This applies as an additional filter to the applications defined in a "
+                    " project's anatomy settings to ignore unavailable applications."
```
Comment (BigRoy) [inline server_addon/applications/server/settings.py:220]:
```suggestion
        description="Enable to show only applications in AYON Launcher"
                    " for which the executable paths are found on the running machine."
                    " This applies as an additional filter to the applications defined in a "
                    " project's anatomy settings to ignore unavailable applications."
```
Should we remove "available" from the first sentence?

Comment (MustafaJafar) [reply server_addon/applications/server/settings.py:220]: I like your suggestion!

**Pull request merged**